assets/main.js.liquid

```diff
@@ -81,33 +81,189 @@ Site.Header = {
 };
 
 Site.Product = {
+  minWidth: 720, // minWidth for fixed header layout
+  numberRelatedToPick: 4, // number of related products to show
   init: function() {
     var _this = this;
 
-    if ($('#product-gallery').length) {
-      _this.setGalleryWidth();
+    _this.$container = $('.container'); // used by toggleFixed() on scroll
+
+    _this.$productHeader = $('#product-header');
+    _this.$productGallery = $('#product-gallery');
+
+    // imagesLoaded hasn't finished yet
+    _this.loadedImages = false;
+
+    if (_this.$productGallery.length) {
+      // We have product images, so we do the fancy layout
+      _this.handleLayout();
     }
 
     if ($('#related-products').length) {
+      // Pick a few Related Products to display
       _this.pickRelated();
     }
   },
 
-  setGalleryWidth: function() {
-    var galleryWidth = 0;
+  handleLayout: function() {
+    var _this = this;
 
-    $('.product-gallery-item').each(function() {
-      galleryWidth += $(this).width();
-    });
+    _this.windowWidth = $(window).width();
+
+    if (_this.windowWidth >= _this.minWidth) {
+      // window is wider than minimum width (720px) for fixed header layout
+
+      if (!_this.loadedImages) {
+        // bind imagesLoaded
+        _this.$productGallery.imagesLoaded( function() {
+          // remove min-height for product gallery holder
+          _this.$productGallery.removeClass('min-height');
+          // size image holder to fit image size
+          _this.sizeImageHolder();
+          // get dimensions for toggleFixed()
+          _this.getFixedHeaderDimensions();
+          // toggle fixed header
+          _this.toggleFixed();
+          // init product image zooming
+          _this.bindZoom();
+          // images are loaded
+          _this.loadedImages = true;
+        });
+      } else {
+        // on window resize
+        // if images are already loaded
+        // do the layout again
+        _this.sizeImageHolder();
```

Comment:
I think this lot should be bound into a `onResize` type function

Reply:
within Site.Product?

Reply:
like Site.Product.onResize()?

Reply:
That is my personal preference for resize events yes. In this case I'd check against `_this.loadedImages` before doing these functions. My preference for the binding of that actual function is to call `Site.Product.onResize()` inside the `Site.onResize()` (but thats just my way with that not sure if sensible or convoluted)

Reply:
i'd just have to check again for the window size and images loaded conditions in the onResize

Reply:
yh fair enough. fine as it is